### DISCUSSION TITLE
Use initialPlacement and overflow before Flip

### DISCUSSION THREAD

**Discussion by IsaacAbrahamson** [floating-ui/floating-ui#2434] [Q&A] [answered]
Hello, I would like to use the Size middleware and `initialPlacement` to place the dropdown below the reference and scroll until some minimum height and then flip the element. 

I copied the middleware from [initial placement](https://floating-ui.com/docs/size#initialplacement) to create an [example codesandbox](https://codesandbox.io/s/floating-ui-react-dom-template-forked-zlkrnz?file=/src/App.js).

In this example, I would like the dropdown to appear at the bottom still but scroll until there is less than `100px` left.

<img width="744" alt="Screenshot 2023-07-12 at 10 58 35 AM" src="https://github.com/floating-ui/floating-ui/assets/17521691/c3baccbb-342f-457d-a96e-773e9fe1c44d">

Is this possible with Size and `initialPlacement`, or will Flip always flip if the element would overflow?

**Comment by atomiks** [comment, marked as answer]
You can choose which one (`size` or `flip`) takes precedence with ordering of the array and some math calcs. https://floating-ui.com/docs/size#initialplacement

Edit: sorry, I see you already linked that :) will check the Sandbox

[This example](https://codesandbox.io/s/floating-ui-react-dom-template-forked-56wr2s?file=/src/App.js) appears to work for me? It will only flip if there's <100px left, and is scrollable.
